### PR TITLE
Remove support for SHOW PARTITIONS statement

### PR DESCRIPTION
**Pull request by electrum**
The Hive $partitions tables should be queried directly instead.

### CODE REVIEW

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java [hunk 1]
```diff
@@ -1283,35 +1279,6 @@ public void testShowColumnsFromPartitions()
...
-
-        assertQueryFails(
-                getSession(),
-                "SHOW PARTITIONS FROM \"" + tableName + "$partitions\"",
-                ".*Table does not have partition columns: .*\\.tpch.test_show_partitions_from_partitions\\$partitions");
-
-        assertQueryFails(
-                getSession(),
-                "SHOW PARTITIONS FROM \"non_existent$partitions\"",
-                ".*Table '.*\\.tpch\\.non_existent\\$partitions' does not exist");
```
Comment (findepi) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java]: Is reading from `non_existent$partitions` covered somewhere else?

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java [hunk 2]
```diff
@@ -1283,35 +1279,6 @@ public void testShowColumnsFromPartitions()
...
-                "WITH (" +
-                "partitioned_by = ARRAY[ 'part1', 'part2' ]" +
-                ") ";
-
-        assertUpdate(getSession(), createTable);
-
-        assertQueryFails(
-                getSession(),
-                "SHOW PARTITIONS FROM \"" + tableName + "$partitions\"",
-                ".*Table does not have partition columns: .*\\.tpch.test_show_partitions_from_partitions\\$partitions");
```
Comment (findepi) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java]: Do we have a test reading from `partitioned_table$partitions$partitions`?

File: presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java [hunk 3]
```diff
@@ -362,18 +361,7 @@ protected Node visitShowPartitions(ShowPartitions showPartitions, Void context)
...
-
-            return simpleQuery(
-                    selectList(new AllColumns()),
-                    new Table(createQualifiedName(partitionsTable)),
-                    showPartitions.getWhere(),
-                    Optional.empty(),
-                    Optional.empty(),
-                    orderBy(showPartitions.getOrderBy()),
-                    showPartitions.getLimit());
+            throw new SemanticException(NOT_SUPPORTED, showPartitions, "SHOW PARTITIONS no longer exists. Use this instead: SELECT * FROM \"%s\"", partitionsTable);
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java]: Commit says:
> Deprecate SHOW PARTITIONS

File: presto-product-tests/src/main/java/com/facebook/presto/tests/convention/TestShowPartitions.java [hunk 4]
```diff
@@ -102,34 +102,36 @@ private static TableDefinition partitionedTableWithVariablePartitionsDefinition(
     public void testShowPartitionsFromHiveTable()
     {
         String tableNameInDatabase = tablesState.get(PARTITIONED_TABLE).getNameInDatabase();
+        String partitionsTable = "\"" + tableNameInDatabase + "$partitions\"";
 
         QueryResult partitionListResult;
 
-        partitionListResult = query("SHOW PARTITIONS FROM " + tableNameInDatabase);
```
Comment (findepi) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/convention/TestShowPartitions.java]: > TestShowPartitions

Consider also renaming the class. (i'm fine if the old name stays though)

File: presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java (hunk 3 above)
Comment (findepi) [reply presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java]: I've read https://github.com/prestodb/presto/pull/10525 now. Fine, but choose different cmt msg, as deprecated functionality usually is one that works (until removed).
Perhaps `Remove support for SHOW PARTITIONS statement` and then later `Remove remainings of SHOW PARTITIONS statement`?

Comment (electrum) [reply presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java]: Agreed, I will update the commit messages.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java (hunk 2 above)
Comment (electrum) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java]: Good idea, I will change this to do that.

File: presto-product-tests/src/main/java/com/facebook/presto/tests/convention/TestShowPartitions.java (hunk 4 above)
Comment (electrum) [reply presto-product-tests/src/main/java/com/facebook/presto/tests/convention/TestShowPartitions.java]: Thoughts on the new name? `TestPartitionsTable`? It's also the only class in the `convention` package, so I suspect it's in the wrong place.

Comment (electrum) [reply presto-product-tests/src/main/java/com/facebook/presto/tests/convention/TestShowPartitions.java]: Will rename to `TestHivePartitionsTable` and move to `hive` package.

**Pull request merged**